driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java

```diff
@@ -378,6 +382,17 @@ private boolean addConnectionIfUnderMaximum() {
         }
     }
 
+    private PooledConnection tryResurrectFromTrash() {
+        long now = System.currentTimeMillis();
+        for (PooledConnection connection : trash)
+            if (connection.maxIdleTime >= now && connection.state.compareAndSet(TRASHED, OPEN)) {
```

Comment:
Should we also consider adding validation here such that the number of available streams on the connection is greater than MIN_AVAILABLE_STREAMS?  That is one criteria for trashing a connection.